zipkintracing/tracing.go

```diff
@@ -0,0 +1,221 @@
+package zipkintracing
+
+import (
+	"fmt"
+	"github.com/labstack/echo/v4/middleware"
+	"net/http"
+	"strconv"
+
+	"github.com/labstack/echo/v4"
+	"github.com/openzipkin/zipkin-go"
+	zipkinhttp "github.com/openzipkin/zipkin-go/middleware/http"
+	"github.com/openzipkin/zipkin-go/model"
+	"github.com/openzipkin/zipkin-go/propagation/b3"
+	"github.com/openzipkin/zipkin-go/reporter"
+	zipkinHttpReporter "github.com/openzipkin/zipkin-go/reporter/http"
+)
+
+type (
+
+	// tracer struct
+	tracer struct {
+		ReportingURL string
+		ServiceName  string
+		tracer       *zipkin.Tracer
+		client       *zipkinhttp.Client
+		reporter     reporter.Reporter
+	}
+
+	// Tracer interface tracing
+	Tracer interface {
```

Comment:
I think creating these huge interfaces are not needed and make testing harder as you need to implement whole interface. In this middleware context I see little need to add it. 

Some good though on interfaces https://www.efekarakus.com/golang/2019/12/29/working-with-interfaces-in-go.html

Reply:
removed interface